control-base/src/main/java/fi/nls/oskari/control/view/GetViewsHandler.java

```diff
@@ -28,88 +31,81 @@
-
+    private Optional<JSONObject> toJSONObject(View view) {
+        try {
             final JSONObject viewJson = new JSONObject();
-            try {
-                viewJson.put(KEY_NAME, view.getName());
-                viewJson.put(KEY_DESCRIPTION, view.getDescription());
-                viewJson.put(KEY_LANG, view.getLang());
-                viewJson.put(KEY_ID, view.getId());
-                viewJson.put(KEY_UUID, view.getUuid());
-                viewJson.put(KEY_ISPUBLIC, view.isPublic());
-                viewJson.put(KEY_ISDEFAULT, view.isDefault());
-                viewJson.put(KEY_PUBDOMAIN, view.getPubDomain());
-                viewJson.put(KEY_URL, view.getUrl());
-                viewJson.put(KEY_METADATA, view.getMetadata());
-                // publisher 2 doesn't need the view info since it loads it using id
-                // The old publisher and normal view listing need them.
-                final JSONObject stateAccu = new JSONObject();
-                for (Bundle bundle : view.getBundles()) {
-
-                    final JSONObject bundleNode = new JSONObject();
-                    try {
-                        bundleNode.put(KEY_STATE, new JSONObject(bundle.getState()));
-                        bundleNode.put(KEY_CONFIG, new JSONObject(bundle.getConfig()));
-                        stateAccu.put(bundle.getBundleinstance(), bundleNode);
-                    } catch (Exception e) {
-                        log.debug("Status " + bundle.getStartup());
-                        log.debug("Config " + bundle.getConfig());
-                    }
+            viewJson.put(KEY_NAME, view.getName());
+            viewJson.put(KEY_DESCRIPTION, view.getDescription());
+            viewJson.put(KEY_LANG, view.getLang());
+            viewJson.put(KEY_ID, view.getId());
+            viewJson.put(KEY_UUID, view.getUuid());
+            viewJson.put(KEY_ISPUBLIC, view.isPublic());
+            viewJson.put(KEY_ISDEFAULT, view.isDefault());
+            viewJson.put(KEY_PUBDOMAIN, view.getPubDomain());
+            viewJson.put(KEY_URL, view.getUrl());
+            viewJson.put(KEY_METADATA, view.getMetadata());
+            viewJson.put(KEY_SRSNAME, view.getSrsName());
+            // publisher 2 doesn't need the view info since it loads it using id
+            // The old publisher and normal view listing need them.
+            final JSONObject stateAccu = new JSONObject();
+            for (Bundle bundle : view.getBundles()) {
+                final JSONObject bundleNode = new JSONObject();
+                try {
+                    bundleNode.put(KEY_STATE, new JSONObject(bundle.getState()));
+                    bundleNode.put(KEY_CONFIG, new JSONObject(bundle.getConfig()));
+                    stateAccu.put(bundle.getBundleinstance(), bundleNode);
+                } catch (Exception e) {
+                    log.debug("Status " + bundle.getStartup());
+                    log.debug("Config " + bundle.getConfig());
```

Comment:
I would change the logging to something like:

    log.warn("View", view.getId(), "has invalid bundle", bundle.getName())

Reply:
Will check, the functionality was copied from current implementation.

Reply:
Didn't change the `Bundle#getState()` or  `Bundle#getConfig()` methods to return `{}` if null. Instead decided to use the JSONHelper methods. They'll .info() the empty values. Checked out how `Bundle` manages the state and config as both String and JSONObject, was a bit confusing so didn't want to touch that here.